docs/general/run-a-node/maintenance-guides/handling-network-upgrades.md

```diff
@@ -1,81 +1,173 @@
 ```
 
-To verify that the upgrade descriptor has been received, grep your node's logs for "received upgrade descriptor" message, e.g.:
-
-> {"caller":"upgrade.go:60","epoch":5102,"level":"info","module":"upgrade","msg":"received upgrade descriptor, scheduling shutdown","name":"testnet-upgrade-2021-03-24","ts":"2021-03-24T09:56:52.944552808Z"}
+and on the error output:
 
-If for any reason you would need to cancel a scheduled pending upgrade, use the `cancel-upgrade` command.
+```text
+panic: upgrade: reached upgrade epoch
+```
 
-Node reaching the upgrade epoch will automatically export network state to a genesis file under the following path:`exports/genesis-<CHAIN-ID>-at-<DUMP-BLOCK-HEIGHT>.json`.
+Then, the `oasis-node genesis dump` command will automatically execute and
+the consensus state will be written into
+`genesis-<CHAIN_ID>-at-<LATEST_HEIGHT>.json` file in the working directory,
+where `CHAIN_ID` is the chain ID of the network and `LATEST_HEIGHT` is the
+round of the last block before the upgrade epoch. This command, depending on the
+size of the state, may take some time to finish.
 
-## Manually exporting network state
+:::info
 
-To dump the state of the network to a genesis file, run:
+To check the network's current height and epoch, run:
 
 ```bash
-oasis-node genesis dump \
-  -a unix:/serverdir/node/internal.sock \
-  --genesis.file /serverdir/etc/genesis_dump.json \
-  --height <HEIGHT-TO-DUMP>
+oasis-node control status -a unix:/serverdir/node/internal.sock
 ```
 
-replacing `<HEIGHT-TO-DUMP>` with the block height we specified.
+and observe the value of the `consensus.latest_height` and
+`consensus.latest_epoch` fields respectively.
 
-:::caution
+:::
 
-You can only run the following command _after_ the `<HEIGHT-TO-DUMP>` block height has been reached on the network.
+## Upgrade Oasis Core
 
-To see the network's current height, run:
+Make sure you upgrade your [Oasis Core](../prerequisites/oasis-node.md)
+binaries to the current version specified in the [Network Parameters].
+
+[Network Parameters]: ../../oasis-network/network-parameters.md
+
+## Update Configuration
+
+If the [Upgrade Log] provides instructions for updating your node's
+configuration, update the `/serverdir/etc/config.yml` file accordingly.
+
+[Upgrade Log]: ../upgrade-log.md
+
+This concludes the seamless upgrade procedure (e.g. [Testnet upgrade
+2022-04-04]) and the new version of Oasis Core can be executed.
+
+[Testnet upgrade 2022-04-04]: ../../foundation/testnet/upgrade-log#2022-04-04-upgrade),
+
+## Update ParaTimes {#update-paratimes}
+
+If you are running either a compute or a client ParaTime node, you will often
+need to upgrade the ParaTime as well. The expected versions of each Paratime
+are stored in the network registry. You can query them by:
 
 ```bash
-oasis-node control status -a unix:/serverdir/node/internal.sock
+oasis-node registry runtime list -v -a unix:/serverdir/node/internal.sock
```

Comment:
Maybe include a pipe to a `jq` command  that filters out a single runtime by ID.